config/config.go

```diff
@@ -196,7 +196,7 @@ type PaymailConfig struct {
 func (p *PaymailConfig) CheckDomain(domain string) error {
 	if p.DomainValidationEnabled {
 		if !slices.Contains(p.Domains, domain) {
```

Comment:
I see that it wasn't added in this specific PR, but I am thinking if extending this config.go file with new "utils" functions working on it's struct is a good idea.

I would consider to move those "utils" method under config package but separated file. 
Maybe not in this specific task, but I wanted to highlight it.